code/modules/religion/rites.dm

```diff
@@ -277,6 +277,7 @@
 		to_chat(user, "<span class='warning'>The soul pool is empty...")
 		new /obj/effect/gibspawner/human/bodypartless(altar_turf)
 		user.visible_message("<span class='warning'>The soul pool was not strong enough to bring forth the undead.")
+		GLOB.religious_sect?.adjust_favor(1500, user) //refund if nobody takes the role
```

Comment:
couldn't one do `adjust_favor(favor_cost, user)` instead of having to set it manually?

Reply:
huh, never really thought about that
It *should* work